src/app/clusters/scenes/SceneTable.h

```diff
@@ -318,21 +330,50 @@ class SceneTable
     virtual CHIP_ERROR RemoveSceneTableEntry(FabricIndex fabric_index, SceneStorageId scene_id)                       = 0;
     virtual CHIP_ERROR RemoveSceneTableEntryAtPosition(FabricIndex fabric_index, SceneIndex scened_idx)               = 0;
 
-    // Iterators
-    using SceneEntryIterator = CommonIterator<SceneTableEntry>;
+    // SceneHandlers
+    virtual CHIP_ERROR RegisterHandler(SceneHandler * handler) = 0;
+    virtual CHIP_ERROR UnregisterHandler(uint8_t position)     = 0;
 
-    virtual SceneEntryIterator * IterateSceneEntry(FabricIndex fabric_index) = 0;
+    // Extension field sets operation
+    virtual CHIP_ERROR SceneSaveEFS(SceneTableEntry & scene)                                    = 0;
+    virtual CHIP_ERROR SceneApplyEFS(FabricIndex fabric_index, const SceneStorageId & scene_id) = 0;
 
     // Fabrics
     virtual CHIP_ERROR RemoveFabric(FabricIndex fabric_index) = 0;
 
+    // Iterators
+    using SceneEntryIterator = CommonIterator<SceneTableEntry>;
+
+    virtual SceneEntryIterator * IterateSceneEntry(FabricIndex fabric_index) = 0;
+
     // Handlers
-    SceneHandler * mHandlers[CHIP_CONFIG_SCENES_MAX_CLUSTERS_PER_SCENES] = { nullptr };
-    uint8_t handlerNum                                                   = 0;
+    virtual bool HandlerListEmpty() { return (handlerNum == 0); }
+    virtual bool HandlerListFull() { return (handlerNum >= kMaxSceneHandlers); }
+    virtual uint8_t GetHandlerNum() { return this->handlerNum; }
 
-protected:
-    const uint8_t mMaxScenePerFabric = kMaxScenePerFabric;
+    SceneHandler * mHandlers[kMaxSceneHandlers] = { nullptr };
+    uint8_t handlerNum                          = 0;
 };
 
+/**
+ * Instance getter for the global SceneTable.
+ *
+ * Callers have to externally synchronize usage of this function.
+ *
+ * @return The global Scene Table
+ */
+SceneTable * GetSceneTable();
+
+/**
+ * Instance setter for the global Scene Table.
+ *
+ * Callers have to externally synchronize usage of this function.
+ *
+ * The `provider` can be set to nullptr if the owner is done with it fully.
+ *
+ * @param[in] provider pointer to the Scene Table global instance to use
+ */
+void SetSceneTable(SceneTable * provider);

```

Comment:
If the SceneTable is set on Server, then this global setter is not required. Suggest just having the scene table set in Server's InitParams